napari_btrack/_tests/test_dock_widget.py

```diff
@@ -1,8 +1,111 @@
-from ..track import track
+from unittest.mock import patch
+
+import napari
+import numpy as np
+import pytest
+from btrack import datasets
+from btrack.config import load_config
+from btrack.datasets import cell_config, particle_config
+
+from ..track import _tracker_config_to_widgets, _widgets_to_tracker_config, track
 
 
 def test_add_widget(make_napari_viewer):
+    """Checks that the track widget can be added inside a dock widget."""
     viewer = make_napari_viewer()
     num_dw = len(list(viewer.window._dock_widgets))
     viewer.window.add_function_widget(function=track)
     assert len(list(viewer.window._dock_widgets)) == num_dw + 1
+
+
+@pytest.fixture
+def track_widget():
+    """Provides an instance of the track widget to test"""
+    return track()
+
+
+@pytest.mark.parametrize("config", [cell_config(), particle_config()])
+def test_config_to_widgets_round_trip(track_widget, config):
+    """Tests that going back and forth between
+    config objects and widgets works as expected.
+    """
+    expected_config = load_config(config)
+    _tracker_config_to_widgets(track_widget, expected_config)
+    actual_config = _widgets_to_tracker_config(track_widget)
```

Comment:
It feels odd that these functions have 'symmetric' names, but one returns something and one doesn't. Perhaps `_update_widget_from_config` would be a better name for the first one?